src/components/modebar/index.js

```diff
@@ -97,18 +122,31 @@ proto.createButton = function (config) {
     button.setAttribute('rel', 'tooltip');
     button.className = 'modebar-btn';
 
-    if (config.attr !== undefined) button.setAttribute('data-attr', config.attr);
-    if (config.val !== undefined) button.setAttribute('data-val', config.val);
-    button.setAttribute('data-title', config.title);
+    button.setAttribute('data-title', config.title || '');
     button.setAttribute('data-gravity', config.gravity || 'n');
-    button.addEventListener('click', function () {
-            config.click.apply(_this, arguments);
+
+    if(config.attr !== undefined) button.setAttribute('data-attr', config.attr);
+
+    var val = config.val;
+    if(val !== undefined) {
+        if(typeof val === 'function') val = val(this.graphInfo);
+        button.setAttribute('data-val', val);
+    }
+
+    var click = config.click;
+    if(typeof click !== 'function') {
+        throw new Error('must provide button \'click\' function in button config');
+    }
+    else {
+        button.addEventListener('click', function(ev) {
+            config.click(_this, ev);
         });
+    }
 
-    button.setAttribute('data-toggle', config.toggle);
+    button.setAttribute('data-toggle', config.toggle || false);
     if(config.toggle) button.classList.add('active');
 
-    button.appendChild(this.createIcon(Plotly.Icons[config.icon]));
+    button.appendChild(this.createIcon(Icons[config.icon || 'tooltip_basic']));
```

Comment:
wouldn't it be better to have some otherwise unused icon as the fallback? maybe a question mark, or a blank square, like you get for missing characters in text?